dask/tests/test_distributed.py

```diff
@@ -277,6 +277,38 @@ def reliable_double(x):
     assert_eq(z, np.ones(10) * 4.0)
 
 
+@pytest.mark.parametrize(
+    "io",
+    [
+        "ones",
+        "zeros",
+        "full",
+    ],
+)
+@pytest.mark.parametrize("fuse", [True, False])
+def test_blockwise_array_creation(c, io, fuse):
+    np = pytest.importorskip("numpy")
+    da = pytest.importorskip("dask.array")
+
+    chunks = (5, 2)
+    shape = (10, 4)
+
+    if io == "ones":
+        darr = da.ones(shape, chunks=chunks)
+        narr = np.ones(shape)
+    elif io == "zeros":
+        darr = da.zeros(shape, chunks=chunks)
+        narr = np.zeros(shape)
+    elif io == "full":
+        darr = da.full(shape, 10, chunks=chunks)
+        narr = np.full(shape, 10)
+
+    darr += 2
+    narr += 2
+    with dask.config.set({"optimization.fuse.active": fuse}):
+        da.assert_eq(darr, narr)
```

Comment:
This test is nice to see : )

All of the `assert_eq` functions use the synchronous scheduler, so I don't think we're testing the blockwise io machinery like we want to here. Two possible options would be to compute `darr` before passing it to `assert_eq`, or add a new `scheduler=` keyword to the `assert_eq` utilities which uses the synchronous scheduler by default.

Reply:
Ah - I realized that the other day while I was debugging, and thought I had added a `darr.compute()` call.  I'll add that back for now.

I also like your suggestion to add a `scheduler=` kwarg to `assert_eq`, but I'm also hoping to keep this PR as simple as I can :)